ivy/stateful/module.py

```diff
@@ -680,7 +680,6 @@ def build(
         self,
         *args,
         from_call=False,
-        device=None,
```

Comment:
I think we can still continue with a `device` argument in the `__init__` that marks some default device which we'd use. 
Also, I think we can have a `model.astype` method as well, just came across [this](https://discuss.pytorch.org/t/is-there-an-efficient-way-to-convert-a-model-with-float32-params-to-float64/130767/2) method in `torch` with the same structure, what do you think?